### PR TITLE
Support variants of structs

### PR DESCRIPTION
**Pull request by andrew-hoff**
Fix for issue #173.

Enables support for storing variants that contain structs.

### CODE REVIEW

**Comment by jsouthworth** [comment]
Thanks for the patch. Can you expand the test cases to cover this as well?

**Comment by andrew-hoff** [comment]
Done!

File: dbus_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package dbus
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```
Comment (jsouthworth) [inline dbus_test.go]: It doesn't seem worth the import overhead to use this library for this test.

File: dbus_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	TestInt int
+	TestStr string
+}
+
+func Test_VariantOfStruct(t *testing.T) {
+	tester := TestStruct{TestInt: 123, TestStr: "foobar"}
+	testerDecoded := []interface{}{123, "foobar"}
+	variant := MakeVariant(testerDecoded)
+	input := []interface{}{variant}
+	output := new(TestStruct)
```
Comment (jsouthworth) [inline dbus_test.go]: The following refactor might read better and avoid the need for the additional library and the numerous dependencies:
```
var output TestStruct
if err := Store(input, &output); err != nil {
    t.Fatal(err)
}
if tester != output {
    t.Fatalf("%v != %v\n", tester, output)
}

**Comment by jsouthworth** [comment]
Thanks!

**Comment by andrew-hoff** [comment]
Np!

**Pull request merged**